### PR TITLE
Update Link.php to fix an issue with invalid URLs

### LINKED ISSUE #33258: Issue with invalid URLs inside products and categories
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

 Inside the method `getLanguageLink` the check for the `$params` is always done wrong as the key is present when a page is not found but the actual value is `false` on the key corresponding to the ID.
That makes the application raise a fatal later on when it search for the link:
`Invalid product vars at line 174 in file classes/Link.php`
`category vars at line 405 in file classes/Link.php

### Expected behavior

The page shouldn't raise a 500 error and return a 404.

### Steps to reproduce

1. The easiest way I found to test this is to install [`SEO Friendly URL Module`](https://github.com/softwaremistri/smseourl) which gonna remove the behavior on top of that method to redirect the user and makes it less complex to reproduce the issue.
2. Then update the URLs patterns to remove the ID (that remove the layer on top of the bug. However the bug is not linked directly to it)
3. Then you can enter any wrong URL in category or product and the problem gonna show up.

### PrestaShop version(s) where the bug happened

1.7.8.9

### PHP version(s) where the bug happened

7.4

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

MBS Design

### PR DESCRIPTION
**Pull request by CrochetFeve0251**
| Questions         | Answers|
| ----------------- | -------------------------------------------------------|
| Branch?           | 8.1.x
| Description?      | Inside the method `getLanguageLink` the check for the `$params` is always done wrong as the key is present when a page is not found but the actual value is `false` on the key corresponding to the ID.<br>That makes the application raise a fatal later on when it search for the link:<br> `Invalid product vars at line 174 in file classes/Link.php`<br> `category vars at line 405 in file classes/Link.php`|
| Type?             | bug fix|
| Category?         | FO|
| BC breaks?        | no|
| Deprecations?     | no|
| How to test?      | The easiest way I found to test this is to install [`SEO Friendly URL Module](https://github.com/softwaremistri/smseourl)` which gonna remove the behavior on top of that method to redirect the user and makes it complex to reproduce the issue.<br>Then you can enter any wrong URL in category or product and problem gonna show up.|
| Fixed ticket?     | Fixes #33258, #33306|
| Sponsor company   | MBS Design|

### CODE REVIEW

**Comment by CrochetFeve0251** [comment]
Logic from #33257 based on 8.1 as requested by @M0rgan01

**Review by Hlavtox — CHANGES_REQUESTED** [review]
Hi, several issues:

1) Bad branch in description
2) Does not fix https://github.com/PrestaShop/PrestaShop/issues/33306, did you test it?
3) As discussed in https://github.com/PrestaShop/PrestaShop/pull/33257 - I am not sure this is a correct fix, because this should return a language variant of the current URL, and even if the current is invalid, it should return it, not suddenly return a link to 404 page or some nonsense link to product controller without a product.

I think that a correct behavior should be that this method should not crash, but return a null.
If the object doesn't exist, the validation should be done earlier.

**Comment by CrochetFeve0251** [comment]
> Hi, several issues:
> 
>     1. Bad branch in description
> 
>     2. Does not fix [FO - Error 500 URL with product id = 0 #33306](https://github.com/PrestaShop/PrestaShop/issues/33306), did you test it?
> 
>     3. As discussed in [Update Link.php to fix an issue with invalid URLs #33257](https://github.com/PrestaShop/PrestaShop/pull/33257) - I am not sure this is a correct fix, because this should return a language variant of the current URL, and even if the current is invalid, it should return it, not suddenly return a link to 404 page or some nonsense link to product controller without a product.
> 
> 
> I think that a correct behavior should be that this method should not crash, but return a null. If the object doesn't exist, the validation should be done earlier.

@Hlavtox Look like on 8.1 the issue is different so I removed my old code and added a new one fixing the #33306 issue with a  more consistent solution.

File: controllers/front/ProductController.php [hunk 1]
```diff
@@ -1348,6 +1352,10 @@ public function getBreadcrumbLinks()
             ];
         }
 
+        if (!$this->product) {
+            return $breadcrumb;
+        }
+
```
Comment (kpodemski) [inline controllers/front/ProductController.php:1358]:
```suggestion
```

There is no need to have this check twice.

**Comment by kpodemski** [comment]
Hello @CrochetFeve0251 

Sorry for the late reply, but we didn't get any notification after you sent the new changes. There's no need to double-check here.

**Comment by matks** [comment]
I close this PR as it has been stale for a long time. Thank you for submitting the PR

**Comment by pelopinxu** [comment]
It looks that someone make a fix for this. I tried and works but I know that needs more than this to be merged. 
https://github.com/PrestaShop/PrestaShop/issues/33306#issuecomment-3571675374

**Pull request closed without merging**